### PR TITLE
runtime: update govmm and move to virtio 1.0 for hotplugs

### PR DESCRIPTION
**Pull request by ganeshmaharaj**

### CODE REVIEW

File: virtcontainers/qemu.go [hunk 1]
```diff
@@ -953,7 +953,7 @@ func (q *qemu) hotplugAddBlockDevice(drive *config.BlockDrive, op operation, dev
 		// PCI address is in the format bridge-addr/device-addr eg. "03/02"
 		drive.PCIAddr = fmt.Sprintf("%02x", bridge.Addr) + "/" + addr
 
-		if err = q.qmpMonitorCh.qmp.ExecutePCIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, addr, bridge.ID, romFile, 0, true, q.arch.runNested()); err != nil {
```
Comment (grahamwhaley) [inline virtcontainers/qemu.go]: reading the commit message - if this is just a test to see how the CI goes - please mark as DNM :-)

**Comment by amshinde** [comment]
@ganeshmaharaj CI looks good. Looks like we can get rid of the disable-modern switch.

**Comment by ganeshmaharaj** [comment]
@grahamwhaley this change good to land?

**Comment by grahamwhaley** [comment]
@ganeshmaharaj - I think you need some formal acks from some virtio aware folks first :-) /cc @devimc @amshinde 
Also, I think @jodh-intel will ask you to format the vendor update commits with the short logs of what has changed...

**Comment by jodh-intel** [comment]
Yes please! ;)

- https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs

**Comment by ganeshmaharaj** [comment]
@grahamwhaley @jodh-intel done and done. :)

File: virtcontainers/qemu.go [hunk 2]
```diff
@@ -953,7 +953,7 @@ func (q *qemu) hotplugAddBlockDevice(drive *config.BlockDrive, op operation, dev
 		// PCI address is in the format bridge-addr/device-addr eg. "03/02"
 		drive.PCIAddr = fmt.Sprintf("%02x", bridge.Addr) + "/" + addr
 
-		if err = q.qmpMonitorCh.qmp.ExecutePCIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, addr, bridge.ID, romFile, true, q.arch.runNested()); err != nil {
+		if err = q.qmpMonitorCh.qmp.ExecutePCIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, addr, bridge.ID, romFile, 0, true, q.arch.runNested()); err != nil {
```
Comment (amshinde) [inline virtcontainers/qemu.go]: Can you add this in a separate commit, as it is not a vendoring change.

File: cli/config/configuration-fc.toml.in [hunk 3]
```diff
@@ -137,7 +137,7 @@ block_device_driver = "@DEFBLOCKSTORAGEDRIVER_FC@"
 # that it is running on top a VMM. This will result in the runtime
 # behaving as it would when running on bare metal.
 # 
-#disable_nesting_checks = true
+disable_nesting_checks = true
```
Comment (amshinde) [inline cli/config/configuration-fc.toml.in]: Looks like this flag was being used just for the ebabling/disabling legacy virtio. If so, we should probably get rid of this flag all together (will help in reducing our configuration bloat as well)
Wdyt @jodh-intel @grahamwhaley

**Comment by ganeshmaharaj** [comment]
@amshinde @devimc disabling the check is causing centos & RHEL to fail even the simplest of commands `kata-runtime list --quiet`. Not sure what is happening there. @chavafg @jcvenegas would you have any clues? Let me try and re-create this locally.

**Comment by ganeshmaharaj** [comment]
@amshinde @devimc @jodh-intel @grahamwhaley  if we want to go down the path of completely removing the virtio 0.9 version support, can we allow this PR to land in it's previous form (https://github.com/kata-containers/runtime/commit/ea40d17fbf5522b3f2c4d0d3a411ca2329d3a59b) and unblock virtiofs memory hotplug. Following that, I can work on clearing out the rest of the code?

**Comment by ganeshmaharaj** [comment]
i tried to re-create the centos issue locally and that seems to work just fine. 
```
Running command '/usr/local/bin/kata-runtime [kata-runtime --log=/tmp/bundle489176700/log run --bundle=/tmp/bundle489176700 --console= --pid-file=/tmp/bundle489176700/pid --detach YmYSQ2ICQcri129TJPgN]'
[kata-runtime --log=/tmp/bundle489176700/log run --bundle=/tmp/bundle489176700 --console= --pid-file=/tmp/bundle489176700/pid --detach YmYSQ2ICQcri129TJPgN]
Timeout: 120 seconds
Exit Code: 0
Stdout:
Stderr:
Running command '/usr/local/bin/kata-runtime [kata-runtime state YmYSQ2ICQcri129TJPgN]'
[kata-runtime state YmYSQ2ICQcri129TJPgN]
Timeout: 120 seconds
Exit Code: 0
Stdout: {
  "ociVersion": "1.0.1-dev",
  "id": "YmYSQ2ICQcri129TJPgN",
  "status": "stopped",
  "pid": 4996,
  "bundle": "/tmp/bundle489176700",
  "annotations": {
    "com.github.containers.virtcontainers.pkg.oci.bundle_path": "/tmp/bundle489176700",
    "com.github.containers.virtcontainers.pkg.oci.config": "{\"ociVersion\":\"1.0.0-rc2-dev\",\"root\":{\"path\":\"rootfs\",\"readonly\":true},\"hostname\":\"runc\",\"mounts\":[{\"destination\":\"/proc\",\"type\":\"proc\",\"source\":\"proc\"},{\"destination\":\"/dev\",\"type\":\"tmpfs\",\"source\":\"tmpfs\",\"options\":[\"nosuid\",\"strictatime\",\"mode=755\",\"size=65536k\"]},{\"destination\":\"/dev/pts\",\"type\":\"devpts\",\"source\":\"devpts\",\"options\":[\"nosuid\",\"noexec\",\"newinstance\",\"ptmxmode=0666\",\"mode=0620\",\"gid=5\"]},{\"destination\":\"/dev/shm\",\"type\":\"tmpfs\",\"source\":\"shm\",\"options\":[\"nosuid\",\"noexec\",\"nodev\",\"mode=1777\",\"size=65536k\"]},{\"destination\":\"/dev/mqueue\",\"type\":\"mqueue\",\"source\":\"mqueue\",\"options\":[\"nosuid\",\"noexec\",\"nodev\"]},{\"destination\":\"/sys\",\"type\":\"sysfs\",\"source\":\"sysfs\",\"options\":[\"nosuid\",\"noexec\",\"nodev\",\"ro\"]},{\"destination\":\"/sys/fs/cgroup\",\"type\":\"cgroup\",\"source\":\"cgroup\",\"options\":[\"nosuid\",\"noexec\",\"nodev\",\"relatime\",\"ro\"]}],\"hooks\":{},\"linux\":{\"resources\":{\"devices\":[{\"allow\":false,\"access\":\"rwm\"}]},\"cgroupsPath\":\"kata\",\"namespaces\":[{\"type\":\"pid\"},{\"type\":\"network\"},{\"type\":\"ipc\"},{\"type\":\"uts\"},{\"type\":\"mount\"}],\"maskedPaths\":[\"/proc/kcore\",\"/proc/latency_stats\",\"/proc/timer_list\",\"/proc/timer_stats\",\"/proc/sched_debug\",\"/sys/firmware\"],\"readonlyPaths\":[\"/proc/asound\",\"/proc/bus\",\"/proc/fs\",\"/proc/irq\",\"/proc/sys\",\"/proc/sysrq-trigger\"]},\"process\":{\"consoleSize\":{\"height\":0,\"width\":0},\"user\":{\"uid\":0,\"gid\":0},\"args\":[\"true\"],\"env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\",\"TERM=xterm\"],\"cwd\":\"/\",\"rlimits\":[{\"type\":\"RLIMIT_NOFILE\",\"hard\":1024,\"soft\":1024}],\"noNewPrivileges\":true,\"capabilities\":{\"Bounding\":null,\"Effective\":null,\"Inheritable\":null,\"Permitted\":null,\"Ambient\":null}}}",
    "com.github.containers.virtcontainers.pkg.oci.container_type": "pod_sandbox"
  }
}
Stderr:
Running command '/usr/local/bin/kata-runtime [kata-runtime list --quiet]'
[kata-runtime list --quiet]
Timeout: 120 seconds
Exit Code: 0
Stdout: YmYSQ2ICQcri129TJPgN

Stderr:
Running command '/usr/local/bin/kata-runtime [kata-runtime delete --force YmYSQ2ICQcri129TJPgN]'
[kata-runtime delete --force YmYSQ2ICQcri129TJPgN]
Timeout: 120 seconds
Exit Code: 0
Stdout:
Stderr:
Running command '/usr/local/bin/kata-runtime [kata-runtime list --quiet]'
[kata-runtime list --quiet]
Timeout: 120 seconds
Exit Code: 0
Stdout:
Stderr:

• [SLOW TEST:8.890 seconds]
state
/home/ganeshma/go/src/github.com/kata-containers/tests/functional/state_test.go:26
  container
  /home/ganeshma/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    with workload [true], timeWait 5
    /home/ganeshma/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
------------------------------

Ran 8 of 8 Specs in 10.586 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 0 Skipped PASS

Ginkgo ran 1 suite in 11.716992429s
Test Suite Passed
bash sanity/check_sanity.sh
```
I am going to say this might be azure vs local VM that i am hitting. i think it would be good to take this change in steps. Get the hotplug unblocked and then sort this out completely.

**Comment by devimc** [comment]
@ganeshmaharaj may be the problem is the host's kernel. CentOS and RHEL use old kernels.

File: cli/config/configuration-fc.toml.in (hunk 3 above)
Comment (jodh-intel) [reply cli/config/configuration-fc.toml.in]: Sounds reasonable to me, but let's get input from @mcastelino who added this feature.

**Comment by jodh-intel** [comment]
Again, /cc @mcastelino for input on https://github.com/kata-containers/runtime/pull/1868#issuecomment-510327822.

**Comment by mcastelino** [comment]
LGTM 

The fact that this works on Azure means the kvm issue has been addresses. The issue may crop up on older kernels in which case the user can opt to disable modern

**Comment by mcastelino** [comment]
FYI this seems to be the kvm fix for the original issue 
https://patchwork.kernel.org/patch/10184241/

**Comment by mcastelino** [comment]
The landed fix
https://github.com/torvalds/linux/commit/d391f1207067268261add0485f0f34503539c5b0

**Pull request merged**